src/udev/scsi_id/scsi_id.c

```diff
@@ -180,9 +181,14 @@ static int get_file_options(const char *vendor, const char *model,
         for (;;) {
                 vendor_in = model_in = options_in = NULL;
 
-                buf = fgets(buffer, MAX_BUFFER_LEN, f);
-                if (buf == NULL)
+                r = read_line(f, MAX_BUFFER_LEN, &buf);
+                if (r < 0) {
```

Comment:
hmm, the existing codebase here is a bit messy, but we probably should log here on error... hence maybe:

```c
retval = log_error_errno(r, …);
```